docs/resources/agent.md

```diff
@@ -116,3 +117,30 @@ Optional:
 - `display_name` (String) The user-facing name of this value.
 - `order` (Number) The order determines the position of agent metadata in the UI presentation. The lowest order is shown first and metadata with equal order are sorted by key (ascending order).
 - `timeout` (Number) The maximum time the command is allowed to run in seconds.
+
+
+<a id="nestedblock--resources_monitoring"></a>
+### Nested Schema for `resources_monitoring`
+
+Optional:
+
+- `memory` (Block Set) The memory monitoring configuration for this agent. (see [below for nested schema](#nestedblock--resources_monitoring--memory))
+- `volume` (Block Set) The volumes monitoring configuration for this agent. (see [below for nested schema](#nestedblock--resources_monitoring--volume))
+
+<a id="nestedblock--resources_monitoring--memory"></a>
+### Nested Schema for `resources_monitoring.memory`
+
+Required:
+
+- `enabled` (Boolean) Enable memory monitoring for this agent.
+- `threshold` (Number) The memory usage threshold in percentage at which to trigger an alert. Value should be between 0 and 100.
+
+
+<a id="nestedblock--resources_monitoring--volume"></a>
+### Nested Schema for `resources_monitoring.volume`
+
+Required:
+
+- `enabled` (Boolean) Enable volume monitoring for this agent.
+- `path` (String) The path of the volume to monitor.
+- `threshold` (Number) The volume usage threshold in percentage at which to trigger an alert. Value should be between 0 and 100.
```

Comment:
This is just a suggestion on naming. The name `volume` seems provider-specific and mostly applies to pods and containers. I have also seen `disks` used for VMs.

What do you guys think about naming it `storage`?

Reply:
"storage" is too generic.

The most logical common name (at least to me) is "volume"; it's a generic term used by both [Windows](https://support.microsoft.com/en-us/windows/disk-management-in-windows-ad88ba19-f0d3-0809-7889-830f63e94405) and [Linux](https://www.baeldung.com/linux/drive-partition-volumes).

I think it's pretty clear what "volume" means in all contexts, and we are designing it to be used with paths (which volumes are mounted at) it'll be a natural UX for operators.